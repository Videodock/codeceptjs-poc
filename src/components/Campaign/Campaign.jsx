import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import * as PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import AppbarTop from '../AppbarTop/AppbarTop';
import FileMusicIcon from '../Icons/FileMusicIcon';
import styles from './Campaign.styles';

const useStyles = makeStyles(styles);

export default function Campaign ({ campaignIdentifier }) {
  const { t }   = useTranslation();
  const classes = useStyles();

  return (
    <React.Fragment>
      <AppbarTop />
      <div className={classes.section}>
        <Typography variant="h1">
          Lorem, ipsum dolor sit amet consectetur adipisicing.
        </Typography>
        <Typography variant="body1">
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Necessitatibus nostrum eveniet accusantium ipsum excepturi reiciendis mollitia? Assumenda soluta doloremque fugit! Repellendus accusantium aliquid aperiam corporis obcaecati quia praesentium minima architecto.
        </Typography>
        <div className={classes.creativeWork}>
          <div className={classes.fileMusicIcon}>
            <FileMusicIcon />
          </div>
          <div>
            <Typography variant="h3">
             Lorem, ipsum dolor sit amet consectetur adipisicing.
            </Typography>
            <Typography variant="body1">
              Lorem, ipsum.
            </Typography>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}

Campaign.propTypes = {
  campaignIdentifier: PropTypes.string,
};
