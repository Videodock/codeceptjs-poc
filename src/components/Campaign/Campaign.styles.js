import { createStyles } from '@material-ui/styles';

export default ({ shape, palette }) => createStyles({
  section: {
    margin  : 44,
    maxWidth: 1440,
  },
  work: {
    display     : 'flex',
    minHeight   : 54,
    marginBottom: 14,
  },
  fileMusicIcon: {
    display       : 'flex',
    justifyContent: 'center',
    alignItems    : 'center',
    margin        : '8px 16px 8px 0',
    '& svg'       : {
      fill  : 'url(#FileMusicIconGradient)',
      height: 36,
      width : 30,
    },
  },
  actions: {
    '& button': {
      marginRight: 16,
    },
    marginBottom: 16,
  },
  shareCampaignInput: {
    padding: '8px 12px',
  },
  shareCampaignAdornedEnd: {
    color       : palette.text.blue,
    borderRadius: shape.borderRadius,
    fontSize    : '13.75px',
    fontWeight  : 'bold',
  },
  shareCampaignInputMarginDense: {
    color     : palette.text.darkGrey,
    fontWeight: 'normal',
  },
  '@media (max-width: 700px)': {
    section: {
      margin      : 16,
      marginBottom: 24,
      '& button'  : {
        width: '100%',
      },
    },
    shareCampaignInputMarginDense: {
      padding: 8,
    },
    textField: {
      width: '100%',
    },
  },
});
