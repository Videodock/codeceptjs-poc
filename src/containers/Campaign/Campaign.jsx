import React from 'react';
import PropTypes from 'prop-types';
import CampaignComponent from '../../components/Campaign';

export default function Campaign ({ campaignIdentifier }) {
  return <CampaignComponent />;
}

Campaign.propTypes = {
  campaignIdentifier: PropTypes.string,
};
