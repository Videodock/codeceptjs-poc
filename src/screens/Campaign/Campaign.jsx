import React from 'react';
import { useTranslation } from 'react-i18next';
import { Helmet } from 'react-helmet';
import CampaignContainer from '../../containers/Campaign';

export default function Campaign ({ match }) {
  const { t }                  = useTranslation('campaign');
  const { campaignIdentifier } = match.params;

  return (
    <React.Fragment>
      <Helmet>
        <title>{t('campaign')}</title>
        <meta name="description" content="" />
        <meta property="og:title" content="Campaign" />
      </Helmet>
      <CampaignContainer campaignIdentifier={campaignIdentifier} />
    </React.Fragment>
  );
}
