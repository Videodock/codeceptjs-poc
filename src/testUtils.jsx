import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
