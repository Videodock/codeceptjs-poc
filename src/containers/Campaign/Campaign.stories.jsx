import React from 'react';
import { storiesOf } from '@storybook/react';
import Campaign from './Campaign';

storiesOf('Campaign', module)
  .add('basic', () => <Campaign />);
