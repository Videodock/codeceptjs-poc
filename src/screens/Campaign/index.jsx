import Campaign from './Campaign';

export * from './Campaign';
export default Campaign;
