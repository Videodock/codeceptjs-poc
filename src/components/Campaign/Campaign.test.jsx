import React from 'react';
import { render } from '../../testUtils';
import Campaign from './Campaign';

describe('<Campaign />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Campaign />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
