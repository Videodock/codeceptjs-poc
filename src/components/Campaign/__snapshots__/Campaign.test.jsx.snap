// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Campaign /> renders and matches snapshot 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-dense makeStyles-dense-10 MuiToolbar-gutters"
  >
    <a
      class="makeStyles-link-12"
      href="/campaign/undefined"
    >
      <img
        alt="Trompa logo"
        class="makeStyles-logo-11"
        src="logo.png"
      />
    </a>
  </div>
</header>
`;
